contracts/hardhat.config.ts

```diff
@@ -0,0 +1,102 @@
+import '@nomiclabs/hardhat-waffle';
+import '@typechain/hardhat';
+import 'hardhat-gas-reporter';
+import 'solidity-coverage';
+
+import './tasks/accounts';
+import './tasks/clean';
+
+import { resolve } from 'path';
+
+import { config as dotenvConfig } from 'dotenv';
+import { HardhatUserConfig } from 'hardhat/config';
+import { NetworkUserConfig } from 'hardhat/types';
+
+dotenvConfig({ path: resolve(__dirname, './.env') });
+
+const chainIds = {
+  ganache: 1337,
+  goerli: 5,
+  hardhat: 31337,
+  kovan: 42,
+  mainnet: 1,
+  rinkeby: 4,
+  ropsten: 3,
+};
+
+// Ensure that we have all the environment variables we need.
+let mnemonic = process.env.MNEMONIC as string;
+if (!mnemonic) {
+  console.warn('Please set your MNEMONIC in a .env file');
+  mnemonic = 'taxi taxi taxi taxi taxi taxi taxi taxi taxi taxi taxi taxi';
```

Comment:
Recommend using `test test test test test test test test test test test junk`, which is the default Hardhat mnemonic. It will be a bit more compatible with things like e.g. the Hardhat optimism plugins if we end up using those